open-widget-framework/test/widget_list_test.js

```diff
@@ -1,31 +1,324 @@
+
+    passThroughProps.renderEditWidgetForm()
+    expect(renderEditWidgetFormSpy.withArgs(dummyWidgetId).callCount).to.equal(1)
+
+  })
+
+  it('makeFormProps constructs an appropriate set of formProps', () => {
+    const wrap = mount(<WidgetList {...dummyProps}/>)
+    const instance = wrap.instance()
+    instance.updateWidgetList(dummyWidgetInstances)
+
+    const updateWidgetListSpy = sinon.spy(instance, 'updateWidgetList')
+    const formProps = instance.makeFormProps()
+
+    expect(formProps.widgetListId).to.equal(dummyWidgetListId)
+    expect(formProps.listLength).to.equal(dummyWidgetInstances.length)
+
+    resetSpyHistory()
+    const dummyUrl = apiPath('widget', dummyWidgetId)
+    formProps.fetchData(dummyUrl)
+    expect(fetchSpy.withArgs(dummyUrl).callCount).to.equal(1)
+
+    const dummyError = 'error'
+    formProps.errorHandler(dummyError)
+    expect(dummyProps.errorHandler.withArgs(dummyError).callCount).to.equal(1)
+
+    formProps.onSubmit(dummyWidgetInstances)
+    expect(updateWidgetListSpy.withArgs(dummyWidgetInstances).callCount).to.equal(1)
+
+    const { Loader } = formProps
+    const loader = mount(<Loader />)
+    expect(loader.exists('.default-loader')).to.equal(true)
+  })
+
+  it('renderWidgetList renders a ListWrapper', () => {
+    const wrap = mount(<WidgetList {...dummyProps}/>)
+    const instance = wrap.instance()
+    instance.updateWidgetList(dummyWidgetInstances)
+
+    const listWrap = mount(instance.renderWidgetList())
+    expect(listWrap.exists('.default-list-wrapper')).to.equal(true)
+    expect(listWrap.props()).to.include(instance.makePassThroughProps())
+    expect(listWrap.props()).to.include(dummyProps.listWrapperProps)
+  })
+
+  it('renderListBody runs render widget once for each widget in widget instances', () => {
+    const wrap = mount(<WidgetList {...dummyProps}/>)
+    const instance = wrap.instance()
+    instance.updateWidgetList(dummyWidgetInstances)
+    const renderWidgetSpy = sinon.spy(instance, 'renderWidget')
+    instance.renderListBody(dummyProps.listWrapperProps)
+
+    expect(renderWidgetSpy.callCount).to.equal(dummyWidgetInstances.length)
+    dummyWidgetInstances.map(instance => {
+      expect(renderWidgetSpy.withArgs(instance, dummyProps.listWrapperProps).callCount).to.equal(1)
     })
   })
+
+  it('renderWidget renders a WidgetWrapper', () => {
+    const wrap = mount(<WidgetList {...dummyProps}/>)
+    const instance = wrap.instance()
+    instance.updateWidgetList(dummyWidgetInstances)
+    const dummyWidgetInstance = dummyWidgetInstances[0]
+    const dummyWidgetId = dummyWidgetInstance.id
+
+    const widgetWrap = mount(instance.renderWidget(dummyWidgetInstance, dummyProps.listWrapperProps))
+    expect(widgetWrap.exists('#widget-' + dummyWidgetId)).to.equal(true)
+    expect(widgetWrap.props()).to.include(dummyWidgetInstance)
+    // TODO: figure out how to check for the passthrough props
```

Comment:
Can this be done similar to the `renderWidgetList renders a ListWrapper` test above?

Reply:
I was able to check for the non-widget specific props and then check for the presence of the four widget specific props. The makePassThroughProps test checks that those props are set to appropriate values by the function so that should cover everything.